### PR TITLE
fix: Team list now aware of org teams

### LINKED ISSUE #9542: [CAL-1957] Owner of the organization can't see the teams he created during onboarding) on /teams page.
* Noticed that the owner of the organization can't see the teams he created(during onboarding) on /teams page. Happening because, we aren't creating the membership of that user with the teams. I think we need to fix it, so that he can add members to the teams later on.
* Also, I don't see any invite sent to the user I added during onboarding

\*Originally posted by @hariombalhara in \*[*https://github.com/calcom/cal.com/issues/9231#issuecomment-1593046893*](https://github.com/calcom/cal.com/issues/9231#issuecomment-1593046893)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1957](https://linear.app/calcom/issue/CAL-1957/owner-of-the-organization-cant-see-the-teams-he-created-during)</sub>

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Teams list now shows correct URL for org sub-teams and created sub-teams now creates membership for owner so they can see the teams in the page.

Fixes #9542 

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

Create an org with sub-teams from wizard and go to Teams page, see the correct URL and that org team per se is not shown.

## Mandatory Tasks
- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/ee/teams/components/TeamListItem.tsx [hunk 1]
```diff
@@ -124,7 +127,11 @@ export default function TeamListItem(props: Props) {
       <div className="ms-3 inline-block truncate">
         <span className="text-default text-sm font-bold">{team.name}</span>
         <span className="text-muted block text-xs">
-          {team.slug ? `${process.env.NEXT_PUBLIC_WEBSITE_URL}/team/${team.slug}` : "Unpublished team"}
+          {team.slug
+            ? orgBranding
+              ? `${orgBranding.slug}.${subdomainSuffix()}/${team.slug}`
+              : `${process.env.NEXT_PUBLIC_WEBSITE_URL}/team/${team.slug}`
+            : "Unpublished team"}
```
Comment (leog) [inline packages/features/ee/teams/components/TeamListItem.tsx:134]: Showing correct URL for org sub-teams

File: packages/trpc/server/routers/viewer/organizations/createTeams.handler.ts [hunk 2]
```diff
@@ -49,15 +49,24 @@ export const createTeamsHandler = async ({ ctx, input }: CreateTeamsOptions) =>
...
+    teamNames.flatMap((name) => {
       if (!duplicatedSlugs.includes(name)) {
-        return { name, parentId: orgId, slug: slugify(name) };
+        return prisma.team.create({
+          data: {
+            name,
+            parentId: orgId,
+            slug: slugify(name),
+            members: {
+              create: { userId: ctx.user.id, role: MembershipRole.OWNER, accepted: true },
```
Comment (leog) [inline packages/trpc/server/routers/viewer/organizations/createTeams.handler.ts:61]: Now associating org owner to the created teams

File: packages/trpc/server/routers/viewer/teams/list.handler.ts [hunk 3]
```diff
@@ -23,9 +24,14 @@ export const listHandler = async ({ ctx }: ListOptions) => {
...
-  return memberships.map(({ team, ...membership }) => ({
-    role: membership.role,
-    accepted: membership.accepted,
-    ...team,
-  }));
+  return memberships
+    .filter((mmship) => {
+      const metadata = teamMetadataSchema.parse(mmship.team.metadata);
+      return !metadata?.isOrganization;
+    })
```
Comment (leog) [inline packages/trpc/server/routers/viewer/teams/list.handler.ts:31]: Filtering out organization teams per se. Tried from prisma query but couldn't make it work:
```js
metadata: {
  path: ["isOrganization"],
  not: true,
},
```

File: packages/trpc/server/routers/viewer/organizations/createTeams.handler.ts [hunk 4]
```diff
@@ -49,15 +49,24 @@ export const createTeamsHandler = async ({ ctx, input }: CreateTeamsOptions) =>
 
   const duplicatedSlugs = existingSlugs.filter((slug) => teamNames.includes(slug));
 
-  await prisma.team.createMany({
-    data: teamNames.flatMap((name) => {
+  await prisma.$transaction(
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/organizations/createTeams.handler.ts:52]: @leog any benefit of using transaction here?

File: packages/trpc/server/routers/viewer/teams/list.handler.ts (hunk 3 above)
Comment (zomars) [reply packages/trpc/server/routers/viewer/teams/list.handler.ts:31]: Good enough for now

**Review by zomars — APPROVED** [review]
LGTM

File: packages/trpc/server/routers/viewer/organizations/createTeams.handler.ts (hunk 4 above)
Comment (leog) [reply packages/trpc/server/routers/viewer/organizations/createTeams.handler.ts:52]: Atomicity mainly, from prisma docs: 
> A database transaction refers to a sequence of read/write operations that are guaranteed to either succeed or fail as a whole.

**Pull request merged**